packages/react/src/components/AccountSettings/DeleteUser/DeleteUser.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+import { deleteUser, translate, getLogger } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { Flex } from '../../../primitives';
+import {
+  DefaultWarning,
+  DefaultError,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { DeleteUserProps, DeleteUserState } from './types';
+
+const logger = getLogger('Auth');
+
+function DeleteUser({
+  onSuccess,
+  onError,
+  handleDelete,
+}: DeleteUserProps): JSX.Element | null {
+  const [state, setState] = React.useState<DeleteUserState>('IDLE');
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  // translations
+  const deleteAccountText = translate('Delete Account');
+
+  const { user, isLoading } = useAuth();
+
+  const startConfirmation = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setState('CONFIRMING');
+  };
+
+  const runDeleteUser = React.useCallback(async () => {
+    setState('DELETING');
+    try {
+      if (handleDelete) {
+        /*
+         * run custom delete handler, if provided. We pass `user` so that
+         * developer can do whichever cleanup with the user object they wish.
+         */
+        await handleDelete(user);
+      } else {
+        // else, run default deleteUser function.
+        await deleteUser();
+      }
+      setState('DONE');
+      onSuccess?.();
+    } catch (e) {
+      const error = e as Error;
+      setState('IDLE');
```

Comment:
Not certain it's super important, but have we considered whether we want to track `ERROR` as the state here in case we want to have differing component behavior post error in the future?

Reply:
I have a bit, IMO it doesn't need to be another state because there are no functional difference between `IDLE` and `ERROR` other than UI layer. I don't foresee needing to do so, but open to thoughts.

Reply:
Update, will make this change for clarity of state/code.

Reply:
[78dd9d0](https://github.com/aws-amplify/amplify-ui/pull/2927/commits/78dd9d003fe5a550deac98a633c5b69494ff21eb)